presto-bigquery/src/main/java/io/prestosql/plugin/bigquery/BigQueryFilterQueryBuilder.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.bigquery;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.spi.connector.ColumnHandle;
+import io.prestosql.spi.predicate.Domain;
+import io.prestosql.spi.predicate.Range;
+import io.prestosql.spi.predicate.TupleDomain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.prestosql.plugin.bigquery.BigQueryColumnHandle.PARTITIONDATE;
+import static io.prestosql.plugin.bigquery.BigQueryColumnHandle.PARTITIONTIME;
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toList;
+
+class BigQueryFilterQueryBuilder
+{
+    private static final String QUOTE = "`";
+    private static final String ESCAPED_QUOTE = "``";
+    private final TupleDomain<ColumnHandle> tupleDomain;
+
+    public static Optional<String> buildFilter(TupleDomain<ColumnHandle> tupleDomain)
+    {
+        return new BigQueryFilterQueryBuilder(tupleDomain).buildFilter();
+    }
+
+    private BigQueryFilterQueryBuilder(TupleDomain<ColumnHandle> tupleDomain)
+    {
+        this.tupleDomain = tupleDomain;
+    }
+
+    private Optional<String> buildFilter()
+    {
+        Optional<Map<ColumnHandle, Domain>> domains = tupleDomain.getDomains();
+        return domains.map(this::toConjuncts)
+                .map(this::concat)
+                .map(filter -> filter
+                        .replace(PARTITIONDATE.getName(), PARTITIONDATE.getName())
+                        .replace(PARTITIONTIME.getName(), PARTITIONTIME.getName()));
```

Comment:
I still don't understand how this is anything other than a no-op in its current form.